### PR TITLE
Add CUDA `11.5` variants to CI axis

### PR DESCRIPTION
**Pull request by ajschmidt8**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

The `nvidia/cuda` version `11.5` images aren't out quite yet, but this PR prepares the `conda-forge` Docker images for when they do come out :slightly_smiling_face:.

### CODE REVIEW

**Comment by leofang** [comment]
It seems 11.5.0 images are uploaded to Docker Hub yesterday. Let's try if it works!

@conda-forge-admin, please restart ci

**Comment by leofang** [comment]
hmm.. no response? How about this?

@conda-forge-admin, please rerender

**Comment by leofang** [comment]
Looks like this repo has a different permission requirement so I can't trigger the bots...

**Comment by jakirkham** [comment]
Yeah I don't think this repo is setup with any of those things

Was going to restart in GH Actions, but it didn't show any option to do that. So just power cycled

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -99,6 +99,11 @@ jobs:
           CUDA_VER: "11.4.2"
           CENTOS_VER: "7"
 
+        - DOCKERIMAGE: linux-anvil-cuda
+          DOCKERTAG: "11.5"
+          CUDA_VER: "11.5.0"
+          CENTOS_VER: "7"
```
Comment (leofang) [inline .github/workflows/ci.yaml:105]:
```suggestion
          CENTOS_VER: "8"
```

File: linux-anvil-cuda/Dockerfile [hunk 2]
```diff
@@ -48,7 +48,7 @@ RUN ldconfig -v 2>/dev/null | grep -v ^$'\t' | cut -f1 -d":" >> /etc/ld.so.conf.
 ADD https://loripsum.net/api /opt/docker/etc/gibberish
 
 # Resolves a nasty NOKEY warning that appears when using yum.
-RUN rpm --import /etc/pki/rpm-gpg/RPM-GPG-KEY-CentOS-${CENTOS_VER}
+RUN find /etc/pki/rpm-gpg/ -type f | xargs -I {} rpm --import "{}"
```
Comment (jakirkham) [inline linux-anvil-cuda/Dockerfile:51]: Needed as this file wasn't present. Also there were a couple others in there that should be imported

**Comment by jakirkham** [comment]
Thanks AJ and Leo! 😄

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (jakirkham) [reply .github/workflows/ci.yaml]: Why was CentOS 8 needed here as opposed to CentOS 7?

cc @kkraus14

**Pull request merged**